<a href="https://colab.research.google.com/github/lauratpaes/Anna-Clara-_-Laura_-Lavinia/blob/main/tarefa12.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Atividade semanal 8

Anna Clara Belmiro Tonial 17097091

Laura Tavares Paes 17097601

Lavinia Ramos Lourenço 17081100

##1. Qual modelo usar?

a. Modelo de Bernoulli, pois esta situação apresenta dois resultados possíveis de fracasso ou sucesso, logo, uma semente germinar pode ser entedido como suesso e não germinar como fracasso.

b. Modelo Binomial, pois esse modelo é utilizado para calcular a probabilidade de determinado evento de sucesso ocorrer em um número determinado de eventos. Nesse caso, as 20 sementes plantadas são o numero fixo de eventos, 0,8 é a probabilidade individual de cada uma das 20 sementes germinarem e, por fim as 15 plantas que germinaram é o número de sucessos desejados.

c. Modelo de Poisson, pois esse modelo é utilizado para calcular a probabilidade de um número de ocorências em um intervalo de tempo ou espaço. Nesse caso, desejamos calcular a probabilidade de ocorrência de uma quantidade de ninhos por hectare a partir de uma média conhecida.


##2. Na população de insetos

a. A distribuição que deve ser utilizada é a binomial, uma vez que temos um número de observações fixo de abelhas (10) e cada animal apresenta dois resultados possíveis: possuir a condição genética ou não possuir a condição genética, além de que a probabilidade de apresentar esta característica é constante (igual a 0.3).



In [1]:
from scipy.stats import binom

n = 10
p = 0.3

# b) Probabilidade de 3 abelhas apresentarem a caracterítica
k_b = 3
prob_exactly_3 = binom.pmf(k_b, n, p)
print(f"b) A probabilidade de exatamente 3 abelhas apresentarem a característica é: {prob_exactly_3:.4f}")

# c) Probabilidade de ao menos 1 abelha apresentar a caracterítica dentre as 10
prob_at_least_1 = 1 - binom.pmf(0, n, p)
print(f"c) A probabilidade de pelo menos 1 das 10 abelhas apresentar a característica é: {prob_at_least_1:.4f}")

b) A probabilidade de exatamente 3 abelhas apresentarem a característica é: 0.2668
c) A probabilidade de pelo menos 1 das 10 abelhas apresentar a característica é: 0.9718


b. A probabilidade de 3 abelhas apresentarem a característica é igual a 0.2668. Este cálculo é feito usando a fórmula:
$$P(X = k) = \binom{n}{k} \cdot p^k \cdot (1-p)^{n-k}$$Onde:

n = 10 (número total de abelhas capturadas)

p = 0,3(probabilidade de uma abelha apresentar a característica)

1 - p = 0,7 (probabilidade de não apresentar)

Queremos calcular P(X = 3), ou seja, k = 3. Então, aplicamos na fórmula:

$$P(X = 3) = \binom{10}{3} \cdot (0,3)^3 \cdot (0,7)^{10-3}$$

Ao resolver, que aqui usamos o código python para auxiliar o processo, resultará em 0.2668.

c. A probabilidade de pelo menos 1 abelha dentre as 10 apresentar a caracterítica é de 0.9718. Seria difícil calcular este evento diretamente, pois o espaço amostral seria muito amplo, então calculares a probabilidade do evento complementar (nenhuma abelha apresentar a condição) e faremos a subtração de 1 menos esta probabilidade, que resultará na probabilidade de interesse.

Assim, teremos que P(X = 0) (k = 0), que será aplicado na fórmula:

$$P(X = 0) = \binom{10}{0} \cdot (0,3)^0 \cdot (0,7)^{10-0}$$

Ao resolver e realizar a subtração, que aqui usamos o código python para auxiliar o processo, resultará em 0.9718.


##3. Na observação ecológica

a. A distribuição mais adequada para este caso é a de Poisson, pois ela modela o número de ocorrências de um evento em um intervalo contínuo (neste caso, o tempo é 1 hora), quando sabemos a taxa média ($\lambda$) em que esses eventos acontecem de forma independente, que no caso é 2 por hora.


In [2]:
from scipy.stats import poisson

lambda_rate = 2

# b) Probabilidade de ocorrerem exatamente 3 encontros em uma hora
k_b = 3
prob_exactly_3_poisson = poisson.pmf(k_b, lambda_rate)
print(f"b) A probabilidade de ocorrerem exatamente 3 encontros em uma hora é: {prob_exactly_3_poisson:.4f}")

# c) Probabilidade de nenhum encontro ocorrer durante uma hora
k_c = 0
prob_none_poisson = poisson.pmf(k_c, lambda_rate)
print(f"c) A probabilidade de nenhum encontro ocorrer durante uma hora é: {prob_none_poisson:.4f}")

# d) Probabilidade de ocorrerem mais de 2 encontros em uma hora
# P(X > 2) = 1 - P(X <= 2) = 1 - (P(X=0) + P(X=1) + P(X=2))
prob_more_than_2_poisson = 1 - poisson.cdf(2, lambda_rate)
print(f"d) A probabilidade de ocorrerem mais de 2 encontros em uma hora é: {prob_more_than_2_poisson:.4f}")

b) A probabilidade de ocorrerem exatamente 3 encontros em uma hora é: 0.1804
c) A probabilidade de nenhum encontro ocorrer durante uma hora é: 0.1353
d) A probabilidade de ocorrerem mais de 2 encontros em uma hora é: 0.3233


b. A probabilidade de ocorrerem exatamente 3 encontros em uma hora é de aproximadamente 0.1805. A fórmula da probabilidade de Poisson é:

$$P(X = k) = \frac{e^{-\lambda} \cdot \lambda^k}{k!}$$

Onde:

\lambda = 2 (taxa média de encontros por hora)

k = 2 = número de encontros desejados

Aplicando na fórmula, teremos que:

$$P(X = 3) = \frac{e^{-2} \cdot 2^3}{3!}$$

Ao resolver, teremos a probabilidade de 0.1804.

c. A probabilidade de nenhum encontro ocorrer durante uma hora é de 0,1353. queremos calcular P(X = 0), ou seja, k = 0 e \lambda = 2. Assim, aplicamos na fórmula:

$$P(X = 0) = \frac{e^{-2} \cdot 2^0}{0!}$$

Este cálculo resultará na probabilidade de 0.1353.

d. A probabilidade de ocorrerem mais de 2 encontros em uma hora é de 0.3233. A condição de mais de 2 encontros resultará em um espaço amostral infinito, então, neste caso, calcularemos a probabilidade do evento complementar (menos de ou igual a dois encontros = 0,1 e 2). Assim, calcularemos a probabilidade destas 3 situações e somares, para em seguida subtrair de 1, que resultará na probabilidade de interesse.

Então, calcularemos:

 P(X = 1) (k = 1):

$$P(X = 1) = \frac{e^{-2} \cdot 2^1}{1!} = \frac{0,135335 \cdot 2}{1} \approx 0,27067$$

P(X = 2) (k = 2):

$$P(X = 2) = \frac{e^{-2} \cdot 2^2}{2!}$$

Já sabemos que X = 0 é igual a 0.1353, pois foi calculado na questão anterior. Então, somamos estas 3 probabilidades e subtraimos este valor de 1, que resulta em 0.3233.

##4. Desafio final

a. Bernoulli, pois, neste caso, queremos saber se um único indivíduo está ou não infectado. Logo, os únicos resultados possíveis são de fracasso ou de sucesso em uma única tentativa.

b. Binomial, pois nesta situação há um número fixo de tentativas independentes (50) e queremos saber o número total de sucessos, ou seja, quantas estão infectadas.

c. A principal diferença entre os modelos é sobre o número de tentativas. Bernoulli trabalha apenas com 1 tentiva, enquanto binomial pode trabalhar com um número maior, mas fixo. Dessa forma, pode-se dizer que a binomial é uma extensão de Bernoulli, somando vários ensaios de Bernoulli.